packages/@aws-cdk/aws-stepfunctions-tasks/lib/emr-add-step.ts

```diff
@@ -0,0 +1,183 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Aws, Stack } from '@aws-cdk/core';
+import { getResourceArn } from './resource-arn-suffix';
+
+/**
+ * The action to take when the cluster step fails.
+ * @see https://docs.aws.amazon.com/emr/latest/APIReference/API_StepConfig.html
+ *
+ * Here, they are named as TERMINATE_JOB_FLOW, TERMINATE_CLUSTER, CANCEL_AND_WAIT, and CONTINUE respectively.
+ *
+ * @default CONTINUE
+ *
+ * @experimental
+ */
+export enum ActionOnFailure {
+  /**
+   * Terminate the Cluster on Step Failure
+   */
+  TERMINATE_CLUSTER = 'TERMINATE_CLUSTER',
+
+  /**
+   * Cancel Step execution and enter WAITING state
+   */
+  CANCEL_AND_WAIT = 'CANCEL_AND_WAIT',
+
+  /**
+   * Continue to the next Step
+   */
+  CONTINUE = 'CONTINUE'
+}
+
+/**
+ * Properties for EmrAddStep
+ *
+ * @experimental
+ */
+export interface EmrAddStepProps {
+  /**
+   * The ClusterId to add the Step to.
+   */
+  readonly clusterId: string;
```

Comment:
Hi @chamcca, one more question about using this.

I will probably want to get `clusterId` out of the create cluster state output and pass it here. Normally this would be done with [passing parameters using paths](https://docs.aws.amazon.com/step-functions/latest/dg/connect-parameters.html#connect-parameters-path) but how could I do that here?

Reply:
you can use either the `sfn.Data.stringAt()` or `sfn.TaskInput.fromDataAt().value` methods. These both take a JSON path parameter. 

take a look at the __/test/emr-add-step.test.ts__ there is an example there that gets the clusterId from "$.ClusterId"

Reply:
thanks!